torchbox/torchbox.com
# BE | Standard Page

[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1362070728)

### Description of Changes Made

With the Standard Page model not changing much from the previous build, this PR

- adds a factory class for the `StandardPage` to aid in writing unit tests.
- adds a few unit tests for the `StandardPage`.
- Updates the `StoryBlock` StreamBlock with the revised `ImageBlock`, removing the old **wide image** and **aligned image** blocks in the process

The `StoryBlock` StreamBlock is also used on the `WorkPage` and `BlogPage`, therefore, in additon to the `core` app where the `StandaPage` model resides, we have migrations in the `blog` and `work` apps.

The process of migrating the **wide image** and **aligned image** blocks to the **image** block was done as follows:

1. Update the `ImageBlock` and add a new `image` field to the `StoryBlock` StreamBlock
2. Create migrations (0fb04d4)
3. Create data migrations (c46c38a) using the [Wagtail StreamField Data Migration Utilities](https://docs.wagtail.org/en/v5.2.2/advanced_topics/streamfield_migrations.html#streamfield-data-migrations)
4. Create final migrations to remove the aligned & wide image fields from the `StoryBlock` StreamBlock (ba96671)

### How to Test

- You should be able to create a new standard page with the new image field. 
  - for the image:
    - when the image is decorative checkbox is ticked, the image alt text should be empty, regardless of whether the alt text is specified or not.
    - when the caption / attribution are specified, they should render on the frontend
    - when the alt text is specified, it is used as the image alt text, otherwise, the image title is used  
- Check for existing standard pages (from the old staging / production) where the body StreamField had wide image or aligned image blocks, and verify that those are now classified as image blocks. Also check this for Blog Pages and Work Pages, as they use the same StoryBlock.

### Screenshots

<details>
  <summary>Standard Page - Before (left) and After (right)</summary>

![image](https://github.com/torchbox/torchbox.com/assets/7713776/1a37a7b4-4677-418b-a8de-848b0432d2c7)


</details>

<details>
  <summary>Work Page - Before (left) and After (right)</summary>

![image](https://github.com/torchbox/torchbox.com/assets/7713776/027028ef-5ce6-43a6-ad06-95bc77199d6d)


</details>

<details>
  <summary>Blog Page - Before (left) and After (right)</summary>


![image](https://github.com/torchbox/torchbox.com/assets/7713776/94209912-1bf8-4c28-a924-84a0d1dca5f2)


</details>

### MR Checklist


#### Unit tests


#### Documentation


#### Browser testing

  - Latest version of Chrome on Linux
  - Latest version of Firefox on Linux

#### Data protection


#### Accessibility


#### Sustainability
